htmlPages/editData.html

```diff
@@ -0,0 +1,291 @@
+                </label>
+                <input type="checkbox" name="hobbies" onclick="updateCheckbox()" id="Writing" value="Writing">
+                <label class="p-0 m-0 mx-2" for="write">
+                    Writing
+                </label>
+                <input type="checkbox" name="hobbies" onclick="updateCheckbox()" id="Dancing" value="Dancing">
+                <label class="p-0 m-0 mx-2" for="dance">
+                    Dancing
+                </label>
+            </label>
+            <div id="hobbiesEmpty" class="m-0 p-0 form-text error">Please select hobbies</div>
+            <div id="hobbiesLess" class="m-0 p-0 form-text error">Please choose minimum 2 hobbies</div>
+            <hr>
+        </div>
+    </div>
+    <div class="row px-5 row-style">
+        <h3 class="p-0 m-0 text-danger">IN CASE OF EMERGENCY PLEASE NOTIFY</h3>
+        <hr>
+        <label class="p-0 m-0" for="name">Name:
+            <input type="text" oninput="validateText(this)" id="name2" class="col-11">
+            <div id="name2Empty" class="m-0 p-0 form-text error">Please enter a name</div>
+            <div id="name2Error" class="m-0 p-0 form-text error">Special characters are not allowed</div>
+        </label>
+        <hr>
+    </div>
+    <div class="row px-5 row-style">
+        <div class="col-6 p-0 m-0">
+            <label class="p-0 m-0" for="mobile">Mobile:
+                <input type="number" oninput="validateNumber(this)" min="1000000000" max="9999999999" id="mob2"
+                    oninvalid="setCustomValidity('Please enter valid mobile number.')"
+                    onchange="try{setCustomValidity('')}catch(e){}" class="mobile2">
+                <div id="mob2Empty" class="m-0 p-0 form-text error">Please enter mobile number</div>
+            </label>
+            <hr style="margin-right: 20px">
+        </div>
+        <div class="col-6 p-0 m-0">
+            <label class="p-0 m-0" for="gender">Gender:
+                <label class="p-0 m-0 mx-2">
+                    <input type="radio" name="gender" id="male" onclick="updateRadio()" class="gender" value="Male">
+                    Male
+                </label>
+                <label class="p-0 m-0 mx-2">
+                    <input type="radio" name="gender" id="female" onclick="updateRadio()" class="gender" value="Female">
+                    Female
+                </label>
+                <div id="genEmpty" class="m-0 p-0 form-text error">Please select a gender</div>
+            </label>
+            <hr style="margin-right: 20px">
+        </div>
+    </div>
+    <div class="row px-5 row-style">
+        <div class="col-6 p-0 m-0">
+            <label class="p-0 m-0" for="mobile">Telephone:
+                <input type="number" oninput="validateNumber(this)" min="1000000000" max="9999999999" id="tel2"
+                    oninvalid="setCustomValidity('Please enter valid telephone.')"
+                    onchange="try{setCustomValidity('')}catch(e){}">
```

Comment:
why you are try catching inline in html file ?

Reply:
This is a necessary JS to make sure that custom validity works fine, onchange of the input the custom validity will be set to null. If not done this way the tooltip will keep showing invalid input even if we provide the right input.

Reply:
try to write js code in js file only